components/txn_types/src/types.rs

```diff
@@ -587,6 +587,87 @@ impl WriteBatchFlags {
     }
 }
 
+/// The position info of the last actual write (PUT or DELETE) of a LOCK record.
+/// Note that if the last change is a DELETE, its LastChange can be either
+/// Exist(which points to it) or NotExist.
+#[derive(Clone, Eq, PartialEq)]
+pub enum LastChange {
+    Unknown,
+    Exist(LastChangePosition), // The pointer may point to a PUT or a DELETE record.
```

Comment:
Consider using the `///` documentation comment above the line? So does the next line.

(btw I'm still kind of afraid that the fact that `Exist` and `NotExist` have commo part might be a little confusing..